PR: https://github.com/odoo/odoo/pull/54234

From: e88fc6222280b1252e55e18cde6046e9454c3ebf
From: Simon Genin (ges)
Date: 2020-07-15 08:37:56

Breaking data model changes score: 3, change matches:
-    amount = fields.Float('Cost')
-    cost_generated = fields.Float('Recurring Cost')
-    amount = fields.Float('Cost')

Total Changes: 10

[IMP] fleet: Improve ux of list views

Small UX improvements on list view and edit text of an empty kanban
screen.

Task ID 2291554

closes odoo/odoo#54234

Signed-off-by: Yannick Tivisse (yti) <yti@odoo.com>

================================= pseudo patch: =================================

--- a/addons/fleet/models/fleet_vehicle.py
+++ b/addons/fleet/models/fleet_vehicle.py
@@ -63,8 +63,8 @@ class FleetVehicle(models.Model):
     odometer = fields.Float(compute='_get_odometer', inverse='_set_odometer', string='Last Odometer',
         help='Odometer measure of the vehicle at the moment of this log')
     odometer_unit = fields.Selection([
-        ('kilometers', 'Kilometers'),
-        ('miles', 'Miles')
+        ('kilometers', 'km'),
+        ('miles', 'mi')
         ], 'Odometer Unit', default='kilometers', help='Unit of the odometer ', required=True)
     transmission = fields.Selection([('manual', 'Manual'), ('automatic', 'Automatic')], 'Transmission', help='Transmission Used by the vehicle')
     fuel_type = fields.Selection([

--- a/addons/fleet/models/fleet_vehicle_cost.py
+++ b/addons/fleet/models/fleet_vehicle_cost.py
@@ -19,7 +19,7 @@ class FleetVehicleLogContract(models.Model):
 
     vehicle_id = fields.Many2one('fleet.vehicle', 'Vehicle', required=True, help='Vehicle concerned by this log')
     cost_subtype_id = fields.Many2one('fleet.service.type', 'Type', help='Cost type purchased with this cost', domain=[('category', '=', 'contract')])
-    amount = fields.Float('Cost')
+    amount = fields.Monetary('Cost')
     date = fields.Date(help='Date when the cost has been executed')
     company_id = fields.Many2one('res.company', 'Company', default=lambda self: self.env.company)
     currency_id = fields.Many2one('res.currency', related='company_id.currency_id')
@@ -45,7 +45,7 @@ class FleetVehicleLogContract(models.Model):
         tracking=True,
         copy=False)
     notes = fields.Text('Terms and Conditions', help='Write here all supplementary information relative to this contract', copy=False)
-    cost_generated = fields.Float('Recurring Cost')
+    cost_generated = fields.Monetary('Recurring Cost')
     cost_frequency = fields.Selection([
         ('no', 'No'),
         ('daily', 'Daily'),
@@ -142,7 +142,7 @@ class FleetVehicleLogServices(models.Model):
 
     active = fields.Boolean(default=True)
     vehicle_id = fields.Many2one('fleet.vehicle', 'Vehicle', required=True, help='Vehicle concerned by this log')
-    amount = fields.Float('Cost')
+    amount = fields.Monetary('Cost')
     description = fields.Char('Description')
     odometer_id = fields.Many2one('fleet.vehicle.odometer', 'Odometer', help='Odometer measure of the vehicle at the moment of this log')
     odometer = fields.Float(compute="_get_odometer", inverse='_set_odometer', string='Odometer Value',
